persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/stringbased/JdbcStringBasedStore.java

```diff
@@ -543,10 +543,12 @@ public boolean contains(Object key) {
          }
          ps.setFetchSize(tableManager.getFetchSize());
          ResultSet rs = ps.executeQuery();
-         return function.apply(rs);
+         return function.apply(rs).doOnComplete(() -> JdbcUtil.safeClose(rs));
       }, kvp -> {
          JdbcUtil.safeClose(kvp.getValue());
-         connectionFactory.releaseConnection(kvp.getKey());
+         Connection connection = kvp.getKey();
+         connection.rollback();
```

Comment:
We need to catch this exception, just to ensure we always return the connection to the pool. Probably just log it similarly to JdbcUtil#safeClose does.